### PR TITLE
#157699757 Finish API endpoint Task and CI integration

### PR DESCRIPTION
**Pull request by holuborhee**
#### What does this PR do?
It implements API endpoints and integrates CI tools for continuous integration

#### Description of Task to be completed?
Have the following endpoints working

`GET api/v1/users/requests`
`POST api/v1/users/requests`
`PUT api/v1/users/requests/:requestId`
`GET api/v1/users/requests/:requestId`

Should contain the following files
- .travis.yml
- .coveralls.yml

#### How should this be manually tested?
After cloning the repo, cd into it and run 
`npm install` and then 
`npm start`

- Using POSTMAN test every endpoint above

#### What are the relevant pivotal tracker stories?
[#157698162](https://www.pivotaltracker.com/story/show/157698162)
[#157697984](https://www.pivotaltracker.com/story/show/157697984)
[#157698542](https://www.pivotaltracker.com/story/show/157698542)
[#157697722](https://www.pivotaltracker.com/story/show/157697722)

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["env", "stage-2"],
+  "plugins": []
+}
```
Comment (benfluleck) [inline .babelrc]: Why is this here?  also, why is the plugins array empty?

File: .coveralls.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+repo_token: F69XcxcGJf5a8H0dFA57giP0ePIzq53nt
```
Comment (benfluleck) [inline .coveralls.yml]: Empty lines need to be added I expect these to have been corrected

File: lib/index.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import RequestController from '../controllers/RequestController';
+
+const app = express();
+
+const port = process.env.PORT || 8082;
+
+// parse application/json and look for raw text
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.text());
```
Comment (benfluleck) [inline lib/index.js]: Why are we using both bodyParser.text and bodyParser.json ?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "maintenance-tracker",
+  "version": "1.0.0",
+  "description": "Helps to track maintenance requests of user",
+  "main": "index.js",
+  "scripts": {
+    "build": "babel lib -d dist",
+    "pretest": "eslint --ignore-path .gitignore .",
+    "serve": "node dist/index.js",
+    "start": "nodemon lib/index.js --exec babel-node",
```
Comment (benfluleck) [inline package.json]: https://codeburst.io/dont-use-nodemon-there-are-better-ways-fc016b50b45e Look for a nodemon alternative

File: package.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "maintenance-tracker",
+  "version": "1.0.0",
+  "description": "Helps to track maintenance requests of user",
+  "main": "index.js",
+  "scripts": {
+    "build": "babel lib -d dist",
+    "pretest": "eslint --ignore-path .gitignore .",
```
Comment (benfluleck) [inline package.json]: what is the purpose of doing this ,can you not just test a specific folder ?

**Review by benfluleck — COMMENTED** [review]
There are too many commits in this branch, I have given you this feedback before. same with the new lines being added to all files. There is a command in user settings that you can use to implement this. Also, I am unable to locate any validations in your project have I missed something?

File: package.json (hunk 5 above)
Comment (holuborhee) [reply package.json]: Currently, I have my folder structure is this way 👇 

image

I want eslint to check all files in each of the folders, except for `node_modules` and `dist` and both files are also ignored by git, so they are in my `.gitignore`, that was why I did the `--ignore-path .gitignore`.

I do not know how testing just a specific folder will work since I need to test more than one folder at this point, unless I restructure the project and put all those in 1 folder, Do you advice that?

File: .coveralls.yml (hunk 2 above)
Comment (holuborhee) [reply .coveralls.yml]: I'm sorry for this that was an oversight.

File: package.json (hunk 4 above)
Comment (holuborhee) [reply package.json]: Ok will check it out and refactor my code

File: .babelrc (hunk 1 above)
Comment (holuborhee) [reply .babelrc]: I added these here, so I will not have to be repeating the `env` and `stage-2` whenever I make use of the babel command in scripts, will remove the empty plugins array

File: lib/index.js (hunk 3 above)
Comment (holuborhee) [reply lib/index.js]: Just removed them now and found out, all my test still passed, I guess I was blindly following a tutorial. I only have this currently. 
```
app.use(bodyParser.urlencoded({ extended: true }));
```

Is that OK?

File: package.json (hunk 5 above)
Comment (holuborhee) [reply package.json]: I have refactored my code a little bit, I am currently testing the `app` and `lib` folder

File: lib/index.js (hunk 3 above)
Comment (benfluleck) [reply lib/index.js]: 👍
